modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk/query_processor.clj

```diff
@@ -217,6 +217,20 @@
   [x]
   (:bigquery-cloud-sdk/temporal-type (meta x)))
 
+(def ^:private temporal-type->supported-units
+  {:timestamp #{:microsecond :millisecond :second :minute :hour :day}
+   :datetime  #{:microsecond :millisecond :second :minute :hour :day :week :month :quarter :year}
+   :date      #{:day :week :month :quarter :year}
+   :time      #{:microsecond :millisecond :second :minute :hour}})
+
+(defmethod temporal-type :+
+  [[_ f [_interval _amount unit]]]
+  (when-let [field-type (temporal-type f)]
+    (if (and (= :timestamp field-type)
```

Comment:
Nitpick but I'd read this as `:timestamp equals field-type` which is a [Yoda condition](https://en.wikipedia.org/wiki/Yoda_conditions)

Reply:
Will reverse.